src/complex/Common/BoundingBox.hpp

```diff
@@ -49,30 +49,56 @@ class BoundingBox
    * @param lowerLeft
    * @param upperRight
    */
-  BoundingBox(const PointType& lowerLeft, const PointType& upperRight)
-  : m_Bounds({lowerLeft[0], lowerLeft[1], lowerLeft[2], upperRight[0], upperRight[1], upperRight[2]})
+  BoundingBox(PointType lowerLeft, PointType upperRight)
+  : m_Lower(lowerLeft)
+  , m_Upper(upperRight)
   {
   }
 
   /**
-   * @brief Constructs a new BoundingBox defined by the array of position values.
+   * @brief Constructs a new BoundingBox3D defined by the array of position values.
    * The format is min X, min Y, min Z, max X, max Y, max Z.
    * @param arr
    */
-  BoundingBox(const std::array<T, 6>& arr)
-  : m_Bounds(arr)
+  template <class PointT = PointType, class = std::enable_if_t<std::is_same<PointT, Point3D<T>>::value>>
+  explicit BoundingBox(const std::array<T, 6>& arr)
+  : m_Lower(Point3D<T>(arr[0], arr[1], arr[2]))
+  , m_Upper(Point3D<T>(arr[3], arr[4], arr[5]))
+  {
+  }
+
+  /**
+   * @brief Constructs a new BoundingBox2D defined by the array of position values.
+   * The format is min X, min Y, max X, max Y.
+   * @param arr
+   */
+  template <class PointT = PointType, class = std::enable_if_t<std::is_same<PointT, Point2D<T>>::value>>
+  explicit BoundingBox(const std::array<T, 4>& arr)
+  : m_Lower(Point2D<T>(arr[0], arr[1]))
+  , m_Upper(Point2D<T>(arr[2], arr[3]))
   {
   }
 
   /**
    * @brief Constructs a new BoundingBox by copying values from the specified pointer.
-   * The pointer is assumed to contain at least six values formatted the same as the
-   * std::array<T, 6> constructor, min X, min Y, min Z, max X, max Y, max Z.
+   * The pointer is assumed to contain either 4 (2D) or 6 (3D) values formatted the same as the
+   * std::array<T, 4> constructor, min X, min Y, max X, max Y or the
+   * std::array<T, 6> constructor, min X, min Y, min Z, max X, max Y, max Z
+   * respectively
    * @param arr
    */
-  BoundingBox(Pointer arr)
-  : m_Bounds({arr[0], arr[1], arr[2], arr[3], arr[4], arr[5]})
+  explicit BoundingBox(nonstd::span<ValueType> arr)
   {
+    if(arr->size() == 6)
+    {
+      m_Lower = Point3D<T>(arr[0], arr[1], arr[2]);
+      m_Upper = Point3D<T>(arr[3], arr[4], arr[5]);
+    }
+    else if(arr->size() == 4)
+    {
+      m_Lower = Point3D<T>(arr[0], arr[1]);
+      m_Upper = Point3D<T>(arr[2], arr[3]);
```

Comment:
Actually looking at this again, shouldn't these in this branch be `Point2D`? And if so, that means we can't branch at runtime since there are different point types.